Svelte 5 migration script: No matching version found for vite-plugin-svelte

### Describe the bug

When attempting to run the migration script on a Svelte 4 repo using pnpm, my dependencies were updated to `vite-plugin-svelte@4.0.0`. However, this release is not valid. It appears the expected version should be `next`, or `4.0.0-next.7`.

### Reproduction

https://github.com/vercel/sveltekit-commerce/pull/33

I was able to fix by manually installing the correct versions for `@sveltejs/vite-plugin-svelte` and `svelte`.

### Logs

```shell
Updated svelte to ^5.0.0
Updated svelte-preprocess to ^6.0.0
Updated @sveltejs/kit to ^2.5.27
Updated @sveltejs/vite-plugin-svelte to ^4.0.0
Updated vite to ^5.4.4
✔ Your project has been migrated

Recommended next steps:

  1: install the updated dependencies ('npm i' / 'pnpm i' / etc) (note that there may be peer dependency issues when not all your libraries officially support Svelte 5 yet. In this case try installing with the --force option)
  2: git commit -m "migration to Svelte 5"
  3: Review the breaking changes at https://svelte-5-preview.vercel.app/docs/breaking-changes

Run git diff to review changes.

▲  sveltekit-commerce/ pn i
 ERR_PNPM_NO_MATCHING_VERSION  No matching version found for @sveltejs/vite-plugin-svelte@^4.0.0

This error happened while installing a direct dependency of /Users/leerob/Developer/sveltekit-commerce

The latest release of @sveltejs/vite-plugin-svelte is "3.1.2".

Other releases are:
  * beta: 2.0.0-beta.3
  * version-2: 2.5.3
  * next: 4.0.0-next.7

If you need the full list of all 107 published versions run "$ pnpm view @sveltejs/vite-plugin-svelte versions".
```


### System Info

```shell
System:
    OS: macOS 14.4
    CPU: (11) arm64 Apple M3 Pro
    Memory: 699.02 MB / 18.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.17.0 - /usr/local/bin/node
    npm: 10.8.2 - /usr/local/bin/npm
    pnpm: 9.10.0 - ~/Library/pnpm/pnpm
    bun: 1.1.29 - ~/.bun/bin/bun
  Browsers:
    Chrome: 129.0.6668.60
    Safari: 17.4
  npmPackages:
    svelte: ^5.0.0 => 4.2.19
```


### Severity

annoyance

--- a/.changeset/shy-hounds-wash.md
+++ b/.changeset/shy-hounds-wash.md
@@ -1,0 +1,5 @@
+---
+'svelte-migrate': patch
+---
+
+fix: use `next` versions for `svelte` and `vite-plugin-svelte`

--- a/packages/migrate/migrations/svelte-5/migrate.js
+++ b/packages/migrate/migrations/svelte-5/migrate.js
@@ -14,12 +14,12 @@ export function update_pkg_json() {
  */
 export function update_pkg_json_content(content) {
 	return update_pkg(content, [
-		['svelte', '^5.0.0'],
+		['svelte', '^5.0.0-next.0'],
 		['svelte-check', '^4.0.0'],
 		['svelte-preprocess', '^6.0.0'],
 		['@sveltejs/enhanced-img', '^0.3.6'],
 		['@sveltejs/kit', '^2.5.27'],
-		['@sveltejs/vite-plugin-svelte', '^4.0.0'],
+		['@sveltejs/vite-plugin-svelte', '^4.0.0-next.0'],
 		[
 			'svelte-loader',
 			'^3.2.3',

--- a/packages/migrate/migrations/svelte-5/migrate.spec.js
+++ b/packages/migrate/migrations/svelte-5/migrate.spec.js
@@ -92,7 +92,7 @@ test('Update package.json', () => {
 	"name": "svelte-app",
 	"version": "1.0.0",
 	"devDependencies": {
-		"svelte": "^5.0.0",
+		"svelte": "^5.0.0-next.0",
 		"svelte-check": "^4.0.0",
 		"svelte-preprocess": "^6.0.0"
 	},

